src/NHibernate/Linq/Visitors/SelectClauseNominator.cs

```diff
@@ -43,11 +44,12 @@ public SelectClauseHqlNominator(VisitorParameters parameters)
 			_parameters = parameters;
 		}
 
-		internal Expression Nominate(Expression expression)
+		internal Expression Nominate(Expression expression, bool isSubQuery = false)
 		{
 			HqlCandidates = new HashSet<Expression>();
 			ContainsUntranslatedMethodCalls = false;
 			_canBeCandidate = true;
+			_isSubQuery = isSubQuery;
 			_stateStack = new Stack<bool>();
 			_stateStack.Push(false);
```

Comment:
Seems additional `_isSubQuery` field is not required with:
```C#
_stateStack.Push(!isSubQuery);
```

Reply:
Indeed. Will fix, and try to avoid the whitespace stuff, while I'm at it.